packages/react-aria-components/docs/Button.mdx

```diff
@@ -159,6 +159,138 @@ A `Button` can be disabled using the `isDisabled` prop.
 
 </details>
 
+## Pending
+
+A `Button` can be put into the pending state using the `isPending` prop.
+Both a `Text` and [ProgressBar](ProgressBar.html) component are required to show the pending state correctly.
+Make sure to internationalize the label you pass to the [ProgressBar](ProgressBar.html) component.
+
+```tsx example
+import {useEffect, useRef, useState} from 'react';
+import {ProgressBar, Text} from 'react-aria-components';
+
+function PendingButton(props) {
+  let [isPending, setPending] = useState(false);
+
+  let timeout = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
+  let handlePress = (e) => {
+    setPending(true);
+    timeout.current = setTimeout(() => {
+      setPending(false);
+      timeout.current = undefined;
+    }, 5000);
+  };
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeout.current);
+    };
+  }, []);
```

Comment:
Since it's just a demo we can probably assume that it's not gonna unmount before the timeout. Nice to keep the example simple.